daemon/api_system_secureboot.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/fdestate"
+)
+
+var systemSecurebootCmd = &Command{
+	// TODO GET returning whether secure boot is relevant for the system?
+
+	Path: "/v2/system-secureboot",
+	POST: postSystemSecurebootAction,
+	WriteAccess: interfaceProviderRootAccess{
+		// TODO find a specialized interface for this, but for now assume that
+		// requests will come only from snaps plugging fwupd interface on the
+		// slot side, which also allows manipulation of EFI variables
+		Interfaces: []string{"fwupd"},
+	},
+}
+
+func postSystemSecurebootAction(c *Command, r *http.Request, user *auth.UserState) Response {
+	contentType := r.Header.Get("Content-Type")
+
+	switch contentType {
+	case "application/json":
+		return postSystemSecurebootActionJSON(c, r)
+	default:
+		return BadRequest("unexpected content type: %q", contentType)
+	}
+}
+
+type securebootRequest struct {
+	Action string `json:"action,omitempty"`
+
+	// Payload is a base64 encoded binary blob, is used in
+	// efi-secureboot-db-prepare action, and carries the DBX update content. The
+	// blob is in the range from few kB to tens of kBs
+	Payload string `json:"payload,omitempty"`
+
+	// DB is used with efi-secureboot-db-prepare action, and indicates the
+	// secureboot keys DB which is a target of the action, possible values are
+	// PK, KEK, DB, DBX
+	DB string `json:"db,omitempty"`
```

Comment:
similarly to the other comment in fdestate this could be named:
```suggestion
	Database string `json:"database,omitempty"`
```